src/core_atmosphere/physics/mpas_atmphys_driver.F

```diff
@@ -226,11 +226,19 @@ subroutine physics_driver(domain,itimestep,xtime_s,mpas_cpl,cells_to_rad_handle,
     ! For all timesteps, we transfer MPAS state to physics
     !
 !$OMP PARALLEL DO
+    if (mpas_cpl%role_includes(ROLE_RADIATION)) then
```

Comment:
If we have MPI ranks doing both radiation and integration, this conditional would lead to the `MPAS_to_physics` routine being called all of the time (and the `MPAS_to_physics_gpu` routine would never be called). Is this what we want to happen?

Reply:
If each rank is running radiation and integration, then shouldn't it  run every timestep? Did I misunderstand this? Or in ROLE03, the ranks iterate through this twice for each timestep?

Reply:
If I'm thinking about this correctly, if we have MPI ranks performing both ROLE_INTEGRATION and ROLE_RADIATION roles, then the if-test in the code, below, will always be true; consequently, the `MPAS_to_physics` routine will always be called by all MPI tasks, and the `MPAS_to_physics_gpu` routine will never be called by any tasks. Since the [MPAS_to_physics_gpu](https://github.com/MPAS-Dev/MPAS-Model/blob/5a7bc4186cf6b377545f08a2784a15fae2fb26b1/src/core_atmosphere/physics/mpas_atmphys_interface.F#L523) routine is being added in this PR, my concern is that there are situations in which this routine would never be called. If we're doing a CPU-only run, then I think the code is correct, but what about future situations where the radiation and integration are both running on GPUs?
```fortran
    if (mpas_cpl%role_includes(ROLE_RADIATION)) then
    do thread=1,nThreads
       call MPAS_to_physics(block%configs,mesh,state,time_lev,diag,diag_physics, &
                            mpas_cpl, cells_to_rad_handle, cells_from_rad_handle, &
                            cellSolveThreadStart(thread), cellSolveThreadEnd(thread))
    end do
    else
    do thread=1,nThreads
       call MPAS_to_physics_gpu(block%configs,mesh,state,time_lev,diag,diag_physics, &
                            mpas_cpl, cells_to_rad_handle, cells_from_rad_handle, &
                            cellSolveThreadStart(thread), cellSolveThreadEnd(thread))
    end do
    end if
```

Reply:
Actually, since the corresponding block of code in the `atmosphere/v6.x-openacc` branch contains the same logic, perhaps we don't need to worry about this right now? I may have confused myself with another discussion where the possibility of running both radiation and integration on the GPUs was mentioned (specifically, [this discussion](https://github.com/MPAS-Dev/MPAS-Model/pull/893#discussion_r677933996) on PR #893).

Reply:
I think you do have a valid point. But yes, we will need make a separate PR when GPU is doing radiation. We can address this at that time. But you can accept this PR with a note that this will require future amendments.

Reply:
Addressing this later (at such time as we have GPU-enabled radiation) sound fine to me. Thanks!